server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/PopulateEmailQueryViewListener.java

```diff
@@ -159,12 +160,16 @@ private Mono<Void> handleAdded(Added added, MessageMetaData messageMetaData, Mai
         MessageId messageId = messageMetaData.getMessageId();
         MailboxId mailboxId = added.getMailboxId();
 
-        return checkMessageStillInOriginMailbox(messageId, session, mailboxId)
-            .filter(FunctionalUtils.identityPredicate())
-            .flatMap(stillInOriginMailbox -> Flux.from(messageIdManager.getMessagesReactive(ImmutableList.of(messageId), FetchGroup.HEADERS, session))
-                .next())
+        Mono<Void> doHandleAdded = Flux.from(messageIdManager.getMessagesReactive(ImmutableList.of(messageId), FetchGroup.HEADERS, session))
+            .next()
             .filter(message -> !message.getFlags().contains(DELETED))
             .flatMap(messageResult -> handleAdded(added.getMailboxId(), messageResult));
+        if (DefaultMailboxes.OUTBOX.equalsIgnoreCase(added.getMailboxPath().getName())) {
```

Comment:
`Role.from(added.getMailboxPath().getName()).contains(Role.OUTBOX)` would even be better...